tests/testthat/test-get_code.R

```diff
@@ -253,13 +253,19 @@ testthat::test_that("detects function usage of the assignment operator", {
     "x <- 1",
     "`<-`(y,x)"
   )
+  code2 <- "`<-`(y, `<-`(x, 2))"
 
   tdata <- eval_code(teal_data(), code)
+  tdata2 <- eval_code(teal_data(), code2)
 
   testthat::expect_identical(
     get_code(tdata, datanames = "y"),
     paste(c(code[1], "y <- x"), collapse = "\n")
   )
+  testthat::expect_identical(
+    get_code(tdata2, datanames = "y"),
+    "y <- x <- 2"
+  )
```

Comment:
Can you add a third example that uses more symbols? (let's say ``"`<-`(y, `<-`(x, `=`(z, 3)))"``)

Which is throwing an error

``` r
code3 <- "`<-`(y, `<-`(x, `=`(z, 3)))"
tdata3 <- teal.code::eval_code(teal.data::teal_data(), code3)
teal.code::get_code(tdata3, datanames = "y")
#> Warning in sym_cond > ass_cond: longer object length is not a multiple of
#> shorter object length
#> Error in if (unique(x$text[ass_cond]) == "->") {: the condition has length > 1
```

<sup>Created on 2024-02-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
@averissimo I think we do not support 2 various assign operator in one line : / and that's a known area
https://github.com/insightsengineering/teal.data/blob/main/R/utils-get_code_dependency.R#L303

Unsure how popular this situation is in the code used for data loading, data preparations, and data aggreagations

Reply:
I would be surprise if this was used by any human

It would be more of a test of `fix_arrows` via public API than `eval_code` itself, but let's not get carried away.

Ignore this one then :smile:

Reply:
> I would be surprise if this was used by any human

Ummm... :speak_no_evil:

Reply:
I will also ignore the below : P

```r
base::`::`("base", "::")("base", "::")("base", "::")("base", "::")("base", "<-")(y, 5)
```

Reply:
:robot: